Sources/TSCBasic/FileSystem.swift

```diff
@@ -12,74 +12,87 @@ import TSCLibc
 import Foundation
 import Dispatch
 
-public enum FileSystemError: Swift.Error {
-    /// Access to the path is denied.
-    ///
-    /// This is used when an operation cannot be completed because a component of
-    /// the path cannot be accessed.
-    ///
-    /// Used in situations that correspond to the POSIX EACCES error code.
-    case invalidAccess
-
-    /// IO Error encoding
-    ///
-    /// This is used when an operation cannot be completed due to an otherwise
-    /// unspecified IO error.
-    case ioError
-
-    /// Is a directory
-    ///
-    /// This is used when an operation cannot be completed because a component
-    /// of the path which was expected to be a file was not.
-    ///
-    /// Used in situations that correspond to the POSIX EISDIR error code.
-    case isDirectory
-
-    /// No such path exists.
-    ///
-    /// This is used when a path specified does not exist, but it was expected
-    /// to.
-    ///
-    /// Used in situations that correspond to the POSIX ENOENT error code.
-    case noEntry
-
-    /// Not a directory
-    ///
-    /// This is used when an operation cannot be completed because a component
-    /// of the path which was expected to be a directory was not.
-    ///
-    /// Used in situations that correspond to the POSIX ENOTDIR error code.
-    case notDirectory
-
-    /// Unsupported operation
-    ///
-    /// This is used when an operation is not supported by the concrete file
-    /// system implementation.
-    case unsupported
-
-    /// An unspecific operating system error.
-    case unknownOSError
-
-    /// File or folder already exists at destination.
-    ///
-    /// This is thrown when copying or moving a file or directory but the destination
-    /// path already contains a file or folder.
-    case alreadyExistsAtDestination
+public struct FileSystemError: Swift.Error, Equatable {
+    public enum Kind: Equatable {
```

Comment:
It's unfortunate to have to split this into a `kind` and a `path`, I think, but adding a `path` to every enum case would be even more unfortunate.